atomic_reactor/dirs.py

```diff
@@ -25,6 +27,10 @@
 logger = logging.getLogger(__name__)
 
 
+def reflink_copy(src, dst, *, follow_symlinks=True):
+    reflink.reflink(str(src), str(dst))
```

Comment:
Does reflink.reflink handle permissions and metadata the same was as [shutil.copy2](https://docs.python.org/3/library/shutil.html#shutil.copy2)?

Reply:
it was done as part of investigation story

reflink itself doesn't have any other parameters, it is just source and dest ....

Reply:
> it was done as part of investigation story

Where? I can't find it

> reflink itself doesn't have any other parameters, it is just source and dest ....

Doesn't need to be reflink params, you can do `src.resolve()`  to follow symlinks (does reflink follow symlinks?), `shutil.copymode()` to copy permissions, `shutil.copystat()` to copy metadata. Though if the copy* functions cause the reflink to be replaced with a regular file, then it would defeat the purpose

Reply:
added resolving also in reflink_copy method (for copytree usecase)

Reply:
Shouldn't we respect the `follow_symlinks` param?